server/controllers/user.ts

```diff
@@ -59,10 +59,14 @@ export function displayRegisterPage(req:Request, res: Response, next: NextFuncti
 }
 
 export function displayUserEditPage(req:Request, res: Response, next: NextFunction): void {
-    if(!req.user)
+
+    const id = req.params.id;
+
+    if(req.user?._id == id )
```

Comment:
I would suggest using `===` instead of `==`. It's just a javascript thing. `===` is the one that acts like `==` in most of the other languages.